content/ADR-253-light-sources.md

```diff
@@ -0,0 +1,117 @@
+- Range: _number_ The maximum distance that can be affected by the light source.
+- Active: _boolean_ Determines if the light is currently on or not.
+
+In lights of type _Spot_, we will also include:
+
+- Inner angle: _number_ This angle, measured from the direction vertex, defines a cone where the light has full intensity. Max 180, can’t be more than outer angle.
+- Outer angle: _number_ This angle, measured from the direction vertex, defines a cone where the light has an effect, the intensity decreases farther away from the inner cone. Max 180.
+
+The `Active` flag lets creators easily turn a light source on or off. We could otherwise achieve the same by setting intensity to 0, but offering a switch makes it easier to retain the prior configuration.
+
+## Shadows
+
+Note: This feature will likely not ba a part of the initial implementation. It's included here to discuss the full vision, but field for this may not be present on the protocol or the SDK until later.
+
+By default lights won’t have shadows. Each light source can chose if they want shadows or not, and if to use hard shadows or soft shadows.
+
+We will add fields for this on the `LightSource` component:
+
+- Shadow type: No shadows / Hard Shadows / Soft shadows
+
+- The creator can chose the shadow resolution as a setting on each light source
+- The shadow resolution is picked by the player in the user’s settings, as a global option. If they have low settings they’ll always use low res
+
+## Limitations
+
+Note: This aspect will likely not ba a part of the initial implementation. It's included here to discuss the full vision, but field for this may not be present on the protocol or the SDK until later. Although restrictions will be applied at an engine level, and each engine could theoretically have different values, it's ideal that we're all aligned on these values, so experiences don't differ accross engines.
+
+We will start with restrictive limitations, and raise them gradually if necessary.
+
+1 light source per parcel. We also need a maximum limit for large scenes.
+What is a good number? TDB while developing and testing.
+
+We should also constrain the maximum brightness, otherwise we could completely blind the player. TDB while developing and testing.
+
+If a light is not being rendered because of going over the limits, the engine should print an error message to console, to make creators aware of the reasons.
+
+## Open world considerations
+
+We advise that each engine only renders the lights of the current scene you’re standing on. In this way:
+
+- Neighboring scenes don’t get affected in performance
+- Neighboring scenes don’t mess with the esthetic of others, at least not when you’re standing on that other scene.
+
+Engines can add a default behavior of fading lights in/out over a period of a second whenever you switch scenes. This is to avoid abrupt light changes when you walk from one parcel to another.
+
+### Affect global ambient light
```

Comment:
The Global Light section needs more detailed specification. Currently, there's a gap between point/spot lights and scene-wide lighting. Consider:

1. **Global Light Components**
A scene's global lighting should be configurable through two complementary components:

```typescript
// Core light properties (when attached to root)
interface Light {
  illuminance: number;  // 400 (sunrise) to 10,000 (midday)
  color: Color3;
  shadows: boolean;
}

// Global light specific properties
interface GlobalLight {
  direction?: Vector3;      // Sunlight direction
  ambientColor?: Color3;    // Sky/environment contribution
  ambientBrightness?: number; // Overall scene brightness multiplier
}
```

2. **Root Entity Behavior**
- `Light` on root = Override default directional (sun) light
- `GlobalLight` on root = Control direction and ambient settings
- Both components can work independently:
  * `Light` alone = Change sun color/intensity
  * `GlobalLight` alone = Change direction/ambient
  * Both = Full control

3. **Default Behavior**
Add specification for default values:
- Default directional light should match typical daylight
- Direction could vary with time-of-day (if implemented)
- Reasonable ambient light for scene visibility

This matches our protocol implementation while providing clear guidance for scene creators and engine implementations.

Reply:
Agreed, I expanded the section, and added more to the initial definition. I however added this as an additional _type_, rather than its own component.